styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java

```diff
@@ -663,7 +663,7 @@ static Workflow parseWorkflowJson(Entity entity, WorkflowId workflowId) throws I
       return OBJECT_MAPPER
           .readValue(entity.getString(PROPERTY_WORKFLOW_JSON), Workflow.class);
     } catch (IOException e) {
-      LOG.error("Failed to read workflow for {}, {}", workflowId.componentId(), workflowId.id());
+      LOG.error("Failed to read workflow for {}, {}", workflowId.componentId(), workflowId.id(), e);
       throw e;
```

Comment:
Interesting, I like to rely on such rethrows ^ to mean that we're promising ourselves to handle the exception with due thoroughness upon catching upstream.